### PR TITLE
Add two robot animals

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
Add two robotic simple mobs : A robot corgi and a robot chicken.
- The Corgi is Robotics' pet as a fully mechanical corgi. Nothing special.
- The chicken, or its full current name 'Soteria Mobile Chemical Transmuter' is a mobile chemical filter & transmuter. When fed various food, it will extract the chemicals inside and output them in bottles. However, some chems will result into special items being layed, including but not restricted to : 
- - Milk to Cheese.
- - Iron to Steel
- - Silver, Plasma, Gold & Uranium to their sheet equivalent.
- - Plasticide to plastic.
- - Silicate to Glass.
- - Nicotine to Cigarette
- Any roboticist can make the SMCT using the board available in Robotics. The board drop upon the death of the chicken, so they can re-use it.

As always a better name and description by @Kazkin would be appreciated.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 1]
```diff
@@ -167,3 +167,57 @@
...
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	colony_friend = TRUE
+	friendly_to_colony = TRUE
+
+/mob/living/simple_animal/corgi/death()
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm:194]:
```suggestion
/mob/living/simple_animal/corgi/E_M/death()
```

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 2]
```diff
@@ -167,3 +167,57 @@
...
+	visible_message("<b>[src]</b> blows apart!")
+	new /obj/effect/decal/cleanable/blood/gibs/robot(src.loc)
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+	s.set_up(3, 1, src)
+	s.start()
+	qdel(src)
+	return
+
+// For repairing damage to the synths.
+/mob/living/simple_animal/corgi/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm:205]:
```suggestion
/mob/living/simple_animal/corgi/E_M/attackby(obj/item/W as obj, mob/user as mob)
```

File: code/modules/mob/living/simple_animal/friendly/corgi.dm [hunk 3]
```diff
@@ -167,3 +167,57 @@
 				for(var/i in list(1,2,4,8,4,2,1,2,4,8,4,2,1,2,4,8,4,2))
 					set_dir(i)
 					sleep(1)
+
+/mob/living/simple_animal/corgi/E_M
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/simple_animal/friendly/corgi.dm:171]: E=Mc^2
?

Comment (r4d6) [reply code/modules/mob/living/simple_animal/friendly/corgi.dm:171]: It is hound that chose the name

File: code/modules/mob/living/simple_animal/friendly/corgi.dm (hunk 2 above)
Comment (Trilbyspaceclone) [reply code/modules/mob/living/simple_animal/friendly/corgi.dm:205]: What about this one

File: code/modules/mob/living/simple_animal/friendly/corgi.dm (hunk 3 above)
Comment (Dopiotl) [reply code/modules/mob/living/simple_animal/friendly/corgi.dm:171]: It's E-N, it reads like Ian, it's a robotic corgi that exists on some other SS13 codebases.

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 4]
```diff
@@ -335,4 +335,142 @@ var/global/chicken_count = 0
...
+				egg_type = /obj/item/reagent_containers/glass/bottle
+
+	if(!egg_type) // In case we don't have something to lay.
+		return FALSE
+
+	var/egg = new egg_type(get_turf(src))
+	if(istype(egg, /obj/item/reagent_containers))
+		var/obj/item/reagent_containers/RC = egg
+		RC.reagents.clear_reagents() // Remove all of the previous chem, just in case
+		RC.reagents.add_reagent(chem, chem_to_egg) // Add the reagent in quantity it is supposed to have.
```
Comment (hex37) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm:464]: Do all the possible objects for egg type need to have a reagent added to them?

Comment (r4d6) [reply code/modules/mob/living/simple_animal/friendly/farm_animals.dm:464]: Only for those that don't spawn anything specific.
If they don't, then it choose to spawn a bottle, filled with the 20 chems that it used up.

File: maps/Nadezhda/map/_Nadezhda_Colony_Omni_STU.dmm [hunk 5]
```diff
@@ -47963,6 +47964,7 @@
 /obj/effect/floor_decal/industrial/warning{
 	dir = 9
 	},
+/mob/living/simple_animal/corgi/E_M,
```
Comment (r4d6) [inline maps/Nadezhda/map/_Nadezhda_Colony_Omni_STU.dmm:47967]:
```suggestion
/mob/living/simple_animal/corgi/E_N,
```

**Pull request merged**